cli/container.go

```diff
@@ -145,13 +145,18 @@ func createContainer(options runvOptions, vm *hypervisor.Vm, container, bundle,
 }
 
 func deleteContainer(vm *hypervisor.Vm, root, container string, force bool, spec *specs.Spec, state *State) error {
-
-	// todo: check the container from vm.ContainerList()
-	// todo: check the process of state.Pid in case it is a new unrelated process
-
 	// non-force killing can only be performed when at least one of the realProcess and shimProcess exited
-	exitedVM := vm.SignalProcess(container, "init", syscall.Signal(0)) != nil // todo: is this check reliable?
-	exitedHost := syscall.Kill(state.Pid, syscall.Signal(0)) != nil
+	exitedVM := true
+	for _, c := range vm.ContainerList() {
+		if c == container {
+			exitedVM = vm.SignalProcess(container, "init", syscall.Signal(0)) != nil // todo: is this check reliable?
+			break
+		}
+	}
+	exitedHost := true
+	if containerShimAlive(state) {
```

Comment:
Is system.Stat() more reliable than syscall.Kill() in which case we don't need syscall.Kill() anymore?